### ISSUE TITLE
Exporting analysis request filtering by client does not work

### ISSUE THREAD

**Issue by ckgathi** [bhp-lims/bhp.lims#229]
## Steps to reproduce
1. Go to clients
2. Select a client
3. AT the end of the url append /lims_export
## Current behavior
1. An error is given
## Expected behavior
1. Export of csv or excel file
## Screenshot (optional)
<img width="924" alt="Screenshot 2019-04-10 at 15 36 26" src="https://user-images.githubusercontent.com/4726008/56013199-e591fd80-5cef-11e9-9df1-c31d3d156e3c.png">

**Comment by xispa** [comment]
Just note that you will not get this error if you stick to the client path directly. It will work if instead of `<site-path>/client-1/analysisrequests/lims_export` you do `<site_path>/client-1/lims_export`

**Comment by ckgathi** [comment]
Please note that this is agent because there are studies that already want their data exported to analyse and or use for other purposes

I have noticed that this link export up to 30 items: https://senaite-server.bhp.org.bw/clients/client-11/lims_export

how do i export all items. 
You did say that one has to add &limit=0

where on the url above is this added?

**Comment by ckgathi** [comment]
When doing an export only certain fields are exported why are all fields not exported and how does one also export results

**Comment by xispa** [comment]
Yes, not all them are there. These are the fields bhp's export thing is displaying:
https://github.com/bhp-lims/bhp.lims/blob/master/bhp/lims/export.py#L87-L116

We added this export.py to show that one could export any data of the system to an xls format easily. We just put there the fields we thought could be interesting to look at. This said, remember the proper way for other apps to gather data from the system is by using the jsonapi. On the other hand, if the clinics want to be able to export data from the system on-demand by their own, I suggest to create a specific report for this, as discussed with @cmaphorisa a while ago.

**Comment by ckgathi** [comment]
This is not that kind of export. May we need to have this discussed in a meeting as i indicated there is data that the lab would want to export and there is data that the clinics would want and there is data that the PIs may want to export and they is data that a statistician would want to export.

I think the system should allow to export all data that is in the system. the system should be flexible in exporting data out of it with allowance to chose what to export because data is capture for the purpose to export and analyse late or to export for other uses.

**Comment by xispa** [comment]
The only thing we we need to know is what are the records and fields you want to export, because there is plenty of data that one might find useful, but also plenty of data that one might find useless. Having useful or useless data will mainly depend on the reason why you want the data to be exported. 

If you were asking that you'd like to know the number of specimens registered on a regular basis grouped by clinic, or you'd like to know the number of analyses performed from any given type within a period of time, or how many results are positive for a given test within a period of time grouped by sex and age.... then we could go ahead easily and we would probably create a report (so anyone could generate this csv/pdf/xlsx/xml or whatever format) and even allow them to apply their own filters on the search (category, date range, clinic, sample type, etc.). This is what we discussed with @cmaphorisa a few weeks ago.

If you were asking for BI capabilities in SENAITE, then I have to say that this is not the purpose of a LIMS/LIS and there is no option other than doing some research for BI apps out there and see if would be possible to integrate with SENAITE.

This being said, regular users can access to the information through SENAITE precisely, so unless there is a specific reason (integration, monthly reports, statistics, etc.), I don't think they would expect the system to allow them to do such a "generalist" export to a csv file.

**Comment by ckgathi** [comment]
We will revert with the different request of export that we need. I juast wanted it to be clear that there are different type of users for the system that will want different sets of data, and like you say they will be only interested in specific sets of data. I would not want a situation we some fields can not be exported because the system can only export others only.

**Comment by Espurna** [comment]
Hi @ckgathi , perfect we will wait for the information. Do not worry, Senaite should be able to export any type of data from the DB.

Note that exporting huge amounts of data from the system (for example all analyses done in one year) will need some "strategies"  to overcome the impact this has in the system. For example, when you request a report from PayPal , the system works on backend and warns the user by email as soon as the report is available to download.
